server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -160,12 +161,25 @@ private static boolean hasAdditionalDependencies(JoinPair joinPair) {
             return false;
         }
         boolean[] hasAdditionalDependencies = {false};
-        FieldsVisitor.visitFields(condition, scopedSymbol -> {
-            RelationName relation = scopedSymbol.relation();
-            if (!relation.equals(joinPair.left()) && !relation.equals(joinPair.right())) {
+
+        // Un-aliased tables
+        RefVisitor.visitRefs(condition, ref -> {
+            RelationName relationName = ref.ident().tableIdent();
+            if (!relationName.equals(joinPair.left()) && !relationName.equals(joinPair.right())) {
                 hasAdditionalDependencies[0] = true;
             }
         });
+
+        // Aliased tables
+        if (hasAdditionalDependencies[0] == false) {
+            FieldsVisitor.visitFields(condition, scopedSymbol -> {
+                RelationName relationName = scopedSymbol.relation();
+                if (!relationName.equals(joinPair.left()) && !relationName.equals(joinPair.right())) {
+                    hasAdditionalDependencies[0] = true;
+                }
+            });
+        }
```

Comment:
Looks like both uses the same consumer, maybe the consumer could be stored and re-used?

Reply:
It's not the same consumer, one is Reference the other ScopedSymbol, so I could extract the if part only, and I think we don't gain much for just 3 lines.